crop/structure.py

```diff
@@ -143,19 +175,125 @@ class ReadingsTinyTagsClass(BASE):
     uptime = Column(Integer, nullable=False)
     battery = Column(Integer, nullable=False)
     validity = Column(Integer, nullable=False)
+    ch0 = Column(Integer, nullable=False)
     ch1 = Column(Integer, nullable=False)
     ch2 = Column(Integer, nullable=False)
     ch3 = Column(Integer, nullable=False)
     opt = Column(Integer, nullable=False)
     co2cozir = Column(Integer, nullable=False)
-    tempsht = Column(Integer, nullable=False)
-    humiditysht = Column(Integer, nullable=False)
+    temperature = Column(Integer, nullable=False)
+    humidity = Column(Integer, nullable=False)
     tempds = Column(Integer, nullable=False)
 
     time_created = Column(DateTime(), server_default=func.now())
     time_updated = Column(DateTime(), onupdate=func.now())
 
 
+class ReadingsTinyTagsClass(BASE):
+    """
+    Class for reading the raw tiny tag data
+    """
+
+    __tablename__ = TINYTAGS_READINGS_TABLE_NAME
+
+    # columns
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    sensor_id = Column(
+        Integer,
+        ForeignKey("{}.{}".format(SENSOR_TABLE_NAME, ID_COL_NAME)),
+        nullable=False,
+    )
+
+    time_stamp = Column(DateTime, nullable=False)
+    temperature = Column(Integer, nullable=False)
+
+    time_created = Column(DateTime(), server_default=func.now())
+    time_updated = Column(DateTime(), onupdate=func.now())
+
+
+class ReadingsEnergyClass(BASE):
+    """
+    Class for reading the energy data
+    (monthly)
+    """
+
+    __tablename__ = ENERGY_READINGS_TABLE_NAME
+
+    # columns
+    id = Column(Integer, primary_key=True, autoincrement=True)
+
+    meter_point = Column(String, nullable=False)
+    period = Column(DateTime, nullable=False)
+    energy = Column(Float, nullable=False)
+
+    time_created = Column(DateTime(), server_default=func.now())
+    time_updated = Column(DateTime(), onupdate=func.now())
+
+class CropDataClass(BASE):
+    """
+    Class for reading the crop data
+    (from google sheets)
+    """
+
+    __tablename__ = CROP_SHEET_TABLE_NAME
+
+    # columns
+    id = Column(Integer, primary_key=True, autoincrement=True)
+
+    crop = Column(String, nullable=False)
+    propagation_date = Column(DateTime, nullable=False)
+    column = Column(Integer, nullable=False)
+    aisle = Column(String, nullable=False)
+    shelf = Column(Integer, nullable=False)
```

Comment:
Is this location? Do you think we could join with the locations table instead?

Reply:
yes! that was my next step. pretty nice join.